queue.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+
+class QueueItem(object):
+    def __init__(self, data, prev_item=None, next_item=None):
+        self.data = data
+        self.next_item = next_item
+        self.prev_item = prev_item
```

Comment:
Do you really need both of these variables? Your queue only moves in one direction.